tests/forms_tests/tests/test_forms.py

```diff
@@ -5329,3 +5422,63 @@ class CustomFrameworkForm(FrameworkForm):
             '<label for="id_name" class="required">Name:</label>'
             '<legend class="required">Language:</legend>',
         )
+
+    @override_settings(FORM_RENDERER="forms_tests.tests.test_forms.CustomRenderer2")
+    def test_custom_boundfield(self):
+        class CustomBoundField(BoundField):
+
+            def label_tag(self, contents=None, attrs=None, label_suffix=None, tag=None):
+                attrs = attrs or {}
+                attrs["class"] = "custom-class"
+                return super().label_tag(contents, attrs, label_suffix, tag)
+
+        class CustomBoundField2(BoundField):
+
+            def label_tag(self, contents=None, attrs=None, label_suffix=None, tag=None):
+                attrs = attrs or {}
+                attrs["class"] = "custom-class-2"
+                return super().label_tag(contents, attrs, label_suffix, tag)
+
+        with self.subTest("settings.FORM_RENDERER is the default"):
+
+            class CustomFrameworkForm(FrameworkForm):
+                template_name = "forms_tests/legend_test.html"
+
+                name = CharField()
+
+            f = CustomFrameworkForm()
+
+            self.assertHTMLEqual(
+                str(f),
+                '<label for="id_name">Name</label>' "<label>Language</label>",
+            )
+        with self.subTest("Field-level takes priority over form-level"):
+
+            class CustomFrameworkForm(FrameworkForm):
+                template_name = "forms_tests/legend_test.html"
+                bound_field_class = CustomBoundField2
+
+                name = CharField(bound_field_class=CustomBoundField)
+
+            f = CustomFrameworkForm()
+
+            self.assertHTMLEqual(
+                str(f),
+                '<label for="id_name" class="custom-class">Name:</label>'
+                '<legend class="custom-class-2">Language:</legend>',
+            )
+
+        with self.subTest("Constructor argument takes priority over class attribute"):
+
+            class CustomFrameworkForm(FrameworkForm):
+                bound_field_class = CustomBoundField
+                template_name = "forms_tests/legend_test.html"
+
+                name = CharField()
+
+            f = CustomFrameworkForm(bound_field_class=CustomBoundField2)
+            self.assertHTMLEqual(
+                str(f),
+                '<label class="custom-class-2" for="id_name">Name:</label>'
+                '<legend class="custom-class-2">Language:</legend>',
+            )
```

Comment:
```suggestion
    @override_settings(
        FORM_RENDERER="forms_tests.tests.test_forms.CustomRendererBoundFieldWithoutColon"
    )
    def test_custom_boundfield(self):
        with self.subTest("settings.FORM_RENDERER is the default"):

            class CustomFrameworkForm(FrameworkForm):
                template_name = "forms_tests/legend_test.html"
                name = CharField()

            f = CustomFrameworkForm()

            self.assertHTMLEqual(
                str(f),
                '<label for="id_name">Name</label><label>Language</label>',
            )
        with self.subTest("Field-level takes priority over form-level"):

            class CustomFrameworkForm(FrameworkForm):
                template_name = "forms_tests/legend_test.html"
                bound_field_class = BoundField

                name = CharField(bound_field_class=BoundFieldWithTwoColons)

            f = CustomFrameworkForm()

            self.assertHTMLEqual(
                str(f),
                '<label for="id_name">Name::</label>'
                '<legend>Language:</legend>',
            )

            f = CustomFrameworkForm(bound_field_class=BoundFieldWithoutColon)

            self.assertHTMLEqual(
                str(f),
                '<label for="id_name">Name::</label>'
                '<label>Language</label>',
            )

        with self.subTest("Constructor argument takes priority over class attribute"):

            class CustomFrameworkForm(FrameworkForm):
                bound_field_class = BoundFieldWithTwoColons
                template_name = "forms_tests/legend_test.html"

                name = CharField()

            f = CustomFrameworkForm(bound_field_class=BoundField)
            self.assertHTMLEqual(
                str(f),
                '<label for="id_name">Name:</label>'
                '<legend>Language:</legend>',
            )
            
        with self.subTest("Custom field with bound_field_class"):

            class CustomField(CharField):
                bound_field_class = BoundFieldWithTwoColons

            class CustomFrameworkForm(FrameworkForm):
                template_name = "forms_tests/legend_test.html"

                name = CustomField()
                comment = CustomField(bound_field_class=BoundField)

            f = CustomFrameworkForm()
            self.assertHTMLEqual(
                str(f),
                '<label for="id_name">Name::</label>'
                '<label>Language</label>'
                '<label for="id_comment">Comment:</label>',
            )
```
I think we can reuse `BoundFieldWithoutColon` here
I also think there's some overlaps between the tests and we can make this cover all cases and we remove `OverrideTests.test_custom_bound_field_custom_renderer` and `FormsTestCase.test_form_custom_bound_field` and `FormsTestCase.test_field_custom_bound_field`

What do you think?

Reply:
How about I put all the tests regarding `BoundField` overriding in a new test class, including [this one](https://github.com/django/django/blob/main/tests/forms_tests/tests/test_forms.py#L2811)?

Or even I create a new `test_boundfield_override.py` to also put the test of overriding `BoundField` at rederer level?

Reply:
Yes I think a new test class makes sense